app/status.go

```diff
@@ -295,6 +295,45 @@ func (a *App) SetStatusAwayIfNeeded(userId string, manual bool) {
 	})
 }
 
+func (a *App) SetStatusDndIfNeeded(userId string, manual bool) {
+	if !*a.Config().ServiceSettings.EnableUserStatuses {
+		return
+	}
+
+	status, err := a.GetStatus(userId)
+
+	if err != nil {
+		status = &model.Status{UserId: userId, Status: model.STATUS_OFFLINE, Manual: manual, LastActivityAt: 0, ActiveChannel: ""}
+	}
+
+	if !manual && status.Manual {
+		return // manually set status always overrides non-manual one
+	}
+
+	if !manual {
+		if status.Status == model.STATUS_DND {
+			return
+		}
+	}
+
+	status.Status = model.STATUS_DND
+	status.Manual = manual
+	status.ActiveChannel = ""
```

Comment:
Do we want to blank out the active channel or can we just leave it as is?

Reply:
I guess leave it as is, removed this.